src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy

```diff
@@ -1369,13 +1407,786 @@ class CookieSyncSpec extends BaseSpec {
+
+        and: "Response shouldn't contain requested bidders due to filter"
+        assert !response.getBidderUserSync(ACEEX)
+    }
+
+    def "PBS cookie sync request shouldn't return any bidder when coop sync off and bidder is invalid"() {
+        given: "Empty cookie sync request body"
+        def cookieSyncRequest = new CookieSyncRequest(coopSync: false, bidders: givenBidder)
+
+        when: "PBS processes cookie sync request without cookies"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should have status 'NO_COOKIE'"
+        assert response.status == NO_COOKIE
+
+        and: "Response shouldn't contain proper"
+        assert response.getBidderStatus().size() == 0
+
+        where:
+        givenBidder << [[BOGUS], []]
+    }
+
+    def "PBS cookie sync request shouldn't return all bidders when coop sync #coopSync and limit param specified"() {
+        given: "Empty cookie sync request body"
+        def cookieSyncRequest = new CookieSyncRequest(coopSync: coopSync, bidders: [APPNEXUS, RUBICON], limit: 1)
+
+        when: "PBS processes cookie sync request without cookies"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should have status 'NO_COOKIE'"
+        assert response.status == NO_COOKIE
+
+        and: "Response shouldn't contain proper"
+        assert response.getBidderStatus().size() == cookieSyncRequest.limit
+
+        where:
+        coopSync << [true, false]
+    }
+
+    def "PBS cookie sync request shouldn't increase amount of bidder then define in request"() {
+        given: "Empty cookie sync request body"
+        def cookieSyncRequest = new CookieSyncRequest().tap {
+            limit = 1
+            bidders = [RUBICON, GENERIC]
+            filterSettings = new FilterSettings(image: new MethodFilter(bidders: [RUBICON, GENERIC], filter: INCLUDE))
+        }
+
+        when: "PBS processes cookie sync request without cookies"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should have status 'NO_COOKIE'"
+        assert response.status == NO_COOKIE
+
+        and: "Response shouldn't contain proper"
+        assert response.getBidderStatus().size() == cookieSyncRequest.limit
+    }
+
+    def "PBS cookie sync request should return all users sync bidders when limit is null"() {
+        given: "Empty cookie sync request body"
+        def cookieSyncRequest = new CookieSyncRequest().tap {
+            limit = null
+            bidders = null
+        }
+
+        when: "PBS processes cookie sync request without cookies"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should have status 'NO_COOKIE'"
+        assert response.status == NO_COOKIE
+
+        and: "Response should contain configured bidders"
+        assert response?.getBidderStatus()?.size() == 8
```

Comment:
not sure that just the number 8 is enough. Maybe we can get it from PBS_CONFIG, or specify what it means.